pkg/minishift/util/password_parse.go

```diff
@@ -0,0 +1,33 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Use to parse oc cluster up -h options.
+
+package util
+
+import (
+	"strings"
+)
+
+// ParseRegistrationPassword will parse password and replace
+// shell special characters with escape
+func ParseRegistrationPassword(password string) string {
+	r := strings.NewReplacer("$", "\\$",
+		`"`, `\"`,
+		"`", "\\`",
+		`\`, `\\`)
```

Comment:
IMO this is not a registration password specific method, but a generic "EscapeStringForSSHUse" method. In which case I would move this into the general _util_ package, outside of _minishift_. This can probably used at other places as well. WDYT?